files/usr/share/cinnamon/applets/cornerbar@cinnamon.org/applet.js

```diff
@@ -154,6 +155,17 @@ class CinnamonBarApplet extends Applet.Applet {
         }
     }
 
+    _on_scroll_event(actor, event) {
+        //switch workspace
+        if (this._did_peek == true && this._peek_timeout_id > 0) {
+            Mainloop.source_remove(this._peek_timeout_id);
```

Comment:
You should set this._peek_timeout_id to 0 here also, as is done in the other event handlers, and I don't think this._did_peek will *ever* be true if this._peek_timeout_id > 0. (See the end of _on_enter). You can probably just check the peek timeout - I'm assuming we just want a pending hover peek to cancel if you scroll.